src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -153,6 +153,26 @@ class FunSpec private constructor(builder: Builder) {
     }
   }
 
+  /**
+   * Returns whether we should emit the return type for given [returnType].
+   *
+   * For return types like [Unit], we can omit emitting the return type, only
+   * if it's not a single expression body.
+   * If it is a single expression body, we want to emit [Unit] so that any type
+   * change in the delegated function doesn't inadvertently carry over to the
+   * delegating function.
+   */
+  private fun emitReturnType(returnType: TypeName?): Boolean {
+    if (returnType != null) {
+      return returnType != Unit::class.asTypeName() || isExpressionBody()
+    }
+    return false
+  }
+
+  private fun isExpressionBody(): Boolean {
```

Comment:
Shouldn't this be used elsewhere to determine behavior as well?

Reply:
The only other place it's required is where you actually want to emit the expressionBody returned from `withoutPrefix()` so no point using it there. https://github.com/square/kotlinpoet/blob/16c8e1e5c6bd2d3cbe513fa60ee114a7c4be4c81/src/main/java/com/squareup/kotlinpoet/FunSpec.kt#L94

We are doing a double check of sorts which feels weird but I couldn't find an easy way to consolidate both.